File: apple/Sources/ReactNativeWatchOS/Modifiers/MultilineTextAlignment.swift
import SwiftUI

/// SwiftUI `.multilineTextAlignment(_:)`. Sets how multi-line text within
/// the view is aligned. Defaults to `.center` for an absent/unrecognized
/// value (matching `RNWAlignmentParser.textAlignment`).
enum RNWMultilineTextAlignmentModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("multilineTextAlignment") { view, params, _ in
            let alignment = RNWAlignmentParser.textAlignment(params.string("value"))
            return AnyView(view.multilineTextAlignment(alignment))
        }
    }
}

File: src/modifiers/multilineTextAlignment.ts
import { createModifier } from './createModifier';

/// SwiftUI `TextAlignment` values.
export type TextAlignment = 'leading' | 'center' | 'trailing';

export interface MultilineTextAlignmentParams {
  value?: TextAlignment;
}

/// SwiftUI `.multilineTextAlignment(_:)`. Sets how multi-line text within
/// the view is aligned. Pass an alignment string directly or an object.
export function multilineTextAlignment(
  params?: MultilineTextAlignmentParams | TextAlignment
) {
  if (typeof params === 'string') {
    return createModifier('multilineTextAlignment', { value: params });
  }
  return createModifier('multilineTextAlignment', params);
}
